server/routes/newspaperArchive.ts

```diff
@@ -0,0 +1,69 @@
+import { Router } from 'express';
+import type { Request, Response } from 'express';
+import fetch from 'node-fetch';
+import { s3ConfigPromise } from '../awsIntegration';
+import { log } from '../log';
+import { withIdentity } from '../middleware/identityMiddleware';
+
+type NewspapersRequestBody = {
+	expires?: number; // defaults to 24 hours
+	'query-string'?: string;
+};
+
+// { url: "https://<subdomain>.newspapers.com/…?tpa=<token>" }
+type NewspapersResponseBody = {
+	url: string;
+};
+
+type NewspaperArchiveConfig = {
+	authKey: string;
+};
+
+function base64(input: string) {
+	return Buffer.from(input).toString('base64');
+}
+
+export const newspaperArchiveConfigPromise: Promise<
+	NewspaperArchiveConfig | undefined
+> = s3ConfigPromise<NewspaperArchiveConfig>('authKey')('newspaper-archive');
+
+let authKey: string;
+
+const router = Router();
+
+router.use(withIdentity(401));
+
+router.get('/auth', async (_req: Request, res: Response) => {
+	const subdomain = 'theguardian';
```

Comment:
it says subdomain but it's actually the user name that's used in the Authorisation header on line 57.
I think it's one of those things that's private rather than secret.  So it could give someone a good starting point for attacks, but they couldn't complete them using that information (unless e.g. the password was accidentally blank or something)
I assume authKey has been generated using a secure key generator so I wouldn't personally be concerned, aside from the noise in the PRs from the scanner.
An easy third option to save extra coding would be to store the authKey in S3 as `theguardian:passwordhere` and rename it to authString or similar.

Reply:
That third option seems like a sensible action 👍